lib/active_merchant/billing/gateways/braintree_blue.rb

```diff
@@ -304,9 +304,9 @@ def merge_credit_card_options(parameters, options)
 
         parameters[:credit_card] ||= {}
         parameters[:credit_card].merge!(:options => valid_options)
-        address = options[:billing_address]&.except(:phone)
-        return parameters if address.nil? || address.values.compact.empty?
-        parameters[:credit_card][:billing_address] = map_address(address)
+        billing_address = map_address(options[:billing_address]) if options[:billing_address]
```

Comment:
Does it make sense to just have `map_address` do something reasonable when passed `nil`? We have this pattern in a few other places, looks like, and I'm wondering if they don't all need the empty check. If they do, it might be simpler to go that route.